app/src/main/java/com/example/todolistproject/view/TaskActivity.kt

```diff
@@ -0,0 +1,243 @@
+                mViewModel.deleteTask(task)
+            }.setNegativeButton(getString(R.string.todo_no), null).create()
+        dialog.show()
+    }
+
+    private fun setTaskCompleted(task: Task) {
+        val isCompleted = task.isCompleted == true
+        task.isCompleted = !isCompleted
+
+        mViewModel.updateTask(task)
+    }
+
+    private fun openDatePickerDialog(editText: TextInputEditText) {
+        val calendar = Calendar.getInstance()
+        val dateSetListener = DatePickerDialog.OnDateSetListener { _, year, month, day ->
+            val date = Calendar.getInstance()
+            date.set(Calendar.YEAR, year)
+            date.set(Calendar.MONTH, month)
+            date.set(Calendar.DAY_OF_YEAR, day)
+            setDateText(editText, date)
+        }
+
+        DatePickerDialog(
+            this,
+            dateSetListener,
+            calendar.get(Calendar.YEAR),
+            calendar.get(Calendar.MONTH),
+            calendar.get(Calendar.DAY_OF_MONTH)
+        ).show()
+    }
+
+    private fun setDateText(editText: TextInputEditText, date: Calendar) {
+        val sdf = SimpleDateFormat("yyyy-MM-dd", Locale.getDefault())
+        editText.setText(sdf.format(date.time))
+    }
+
+    fun openListFilter() {
+        val items = arrayOf(TASK, INCOMPLETE, COMPLETED)
+        val checkedItem = items.indexOf(currentFilter)
+        var tempFilter = currentFilter
+        MaterialAlertDialogBuilder(this)
+            .setTitle("Filter")
+            .setNegativeButton("Cancel") { dialog, _ ->
+                dialog.dismiss()
+            }
+            .setPositiveButton("OK") { dialog, _ ->
+                currentFilter = tempFilter
+                showTaskList(false)
+                setFilteredList()
+                dialog.dismiss()
+            }
+            .setSingleChoiceItems(items, checkedItem) { _, index ->
+                tempFilter = items[index]
+            }
+            .show()
+    }
+
+    private fun setFilteredList() {
+        when (currentFilter) {
+            TASK -> {
+                mViewModel.getTaskList().observe(this, {
+                    mAdapter.setTaskItems(it)
+                })
+            }
+            INCOMPLETE -> {
+                mViewModel.getIncompleteTaskList().observe(this, {
+                    mAdapter.setTaskItems(it)
+                })
+            }
+            COMPLETED -> {
+                mViewModel.getCompletedTaskList().observe(this, {
+                    mAdapter.setTaskItems(it)
+                })
+            }
+        }
+        showTaskList(true)
+    }
+
+    private fun showTaskList(visible: Boolean) {
+        if (visible) {
+            binding.progressIndicator.visibility = View.GONE
+            binding.rvTodoList.visibility = View.VISIBLE
+        } else {
+            binding.rvTodoList.visibility = View.GONE
+            binding.progressIndicator.visibility = View.VISIBLE
+        }
+    }
```

Comment:
```suggestion
    private fun showTaskList(visible: Boolean) {
            binding.progressIndicator.isGone = visible
            binding.rvTodoList.isVisible = visible
    }
```

ktx 함수를 사용하면 많이 2줄만으로 표현이 가능합니다.